### PR TITLE
partial import of workshops 

### PR DESCRIPTION
**Pull request by patricoferris**
This PR sets up ood ready for others to do the rest of data importing for the workshops (we may have some help on this). Before that, we need to merge the types etc. first so the job is more data-mangling than ocaml-hacking.

I could do with your input on two questions @tmattio: 
  1. This is one of the first data types that could really benefit from relational information with other things in ood. In particular the "presentations" could all have optional videos, slides and papers linked. Do you think the best way to do this is to (a) store the slug of each item in the yaml and then for ood-gen reconstruct the appropriate data by reading and searching for the entries and copying them in (seems a little space wasteful, maybe it could just double-check they exist so users of ood are guaranteed to be able to reconstruct them). 
  2. A lot of the older workshops did a fantastic job of preserving slides and can be found [in this repository](https://github.com/ocaml/ocaml.org-media), how should we do this now, submodule that repo in here or just having hard links to the slides here on Github ?
  
Depending on how we resolve these issues I'll update this PR accordingly :))

### CODE REVIEW

**Comment by patricoferris** [comment]
Ah mdx got a fix for the error printing https://github.com/realworldocaml/mdx/blob/main/CHANGES.md#fixed :)) I'll update and constrain mdx to `>= 1.10.0`.

**Comment by tmattio** [comment]
> This is one of the first data types that could really benefit from relational information with other things in ood.

Is it possible not to have relationships between the data? I suspect this will introduce a lot of complexity that we'll have to maintain forever, so I'd rather be 100% certain there isn't an alternative solution..

For instance, could we keep the blog posts and slides as subfields of the workshops? If really there's a need to aggregate all of the presentations, we can delegate that on the frontend and ask it to go through the different kinds of data to look for presentations. Having no relationships does not mean that we'll have to duplicate code either: we can reuse a specific type for the presentations across the board and have a single way to process them

**Comment by tmattio** [comment]
> A lot of the older workshops did a fantastic job of preserving slides and can be found in this repository

I'd be wary of having multiple sources of truth for the data. Ideally, I think we'd migrate these in `ood` and ask the maintainers of `ocaml.org-media` to deprecate the repository and redirect to `ood`.

**Comment by patricoferris** [comment]
>For instance, could we keep the blog posts and slides as subfields of the workshops? If really there's a need to aggregate all of the presentations, we can delegate that on the frontend and ask it to go through the different kinds of data to look for presentations. Having no relationships does not mean that we'll have to duplicate code either: we can reuse a specific type for the presentations across the board and have a single way to process them

Reusing types sounds good, the tradeoff I guess is between the complexity in the code vs. duplication in the _data_. Reusing types probably means that data will exist in two different places. For instance we might have a video in `videos.yml` and then the same video in `workshops/en/2020.md`, with this approach the two may be the same but we'll have no way of knowing so if one is updated the other might not be so there is still ongoing complexity with this approach.

For the sake of getting this merged, however, I'm happy to go with this approach and we can always revisit and refactor in the future :)) Thanks for the feedback

File: data/workshops/en/2017.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+  ([extended abstract](https://www.irif.fr/~gradanne/papers/tyre/abstract.pdf)
+   ([local copy](extended-abstract__2017__gabriel-radanne__tyre-typed-regular-expressions.pdf)))
+
+- _Jbuilder: a modern approach to OCaml development_ —
+  Jeremie Dimino, Mark Shinwell  
+  ([local copy](extended-abstract__2017__jeremie-dimino_mark-shinwell__jbuilder-a-modern-approach-to-ocaml-development.pdf))
+
+
+Call for presentations (past)
+=============================
```
Comment (tmattio) [inline data/workshops/en/2017.md]: Could we replace this with standard markdown titles (i.e. the style of the titles we use for the other data)?

File: src/ood-gen/lib/workshop.ml [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        online = metadata.online;
+        important_dates = metadata.important_dates;
+        presentations = metadata.presentations;
+        program_committee = metadata.program_committee;
+        organising_committee = metadata.organising_committee;
+        toc_html = Omd.to_html (Omd.toc omd);
+        body_md = body;
+        body_html = Omd.to_html omd;
+      })
+    "workshops/en/*.md"
```
Comment (tmattio) [inline src/ood-gen/lib/workshop.ml]: Is there a need to have the workshops translated one day?

**Review by tmattio — APPROVED** [review]
Some minor comments, but LGTM!

I'm curious, how did you find the workflow with `ood-gen` + `ood-preview`?

**Comment by patricoferris** [comment]
> I'm curious, how did you find the workflow with ood-gen + ood-preview?

It's pretty slick (we need to write a contributing doc because at a first glance it's a little confusing). I think we can fold `ood-gen` into `ood-cli` as a `gen` subcommand so that we're only producing one executable binary that can lint, gen, fetch blogs, soon fetch videos etc. what do you think ?

I'm also concerned that we have essentially the same types everywhere which is a little laborious when implementing e.g: 
  -  https://github.com/ocaml/ood/blob/main/src/ood-gen/lib/tutorial.ml
  - https://github.com/ocaml/ood/blob/main/src/ood-cli/data.ml#L199
  - https://github.com/ocaml/ood/blob/main/src/ood/ood.mli#L160
  
 I think combining ood-gen and ood-cli should help fix some of this because `type metadata` in gen is identical to `t` in `ood-cli/data` so they *should* be shared I think. I'm also probably going to remove the netlify-cms stuff. The two combined should simplify the repo a good bit. What are your thoughts? :))

**Comment by tmattio** [comment]
> What are your thoughts?

Merging all the tooling into the same CLI seems like a sensible simplification 🙂 

> I'm also concerned that we have essentially the same types everywhere which is a little laborious when implementing

Right, I find it problematic as well, but didn't find a satisfying solution. Do you have any idea how we could reuse this at different places (especially the code generation part)?

File: src/ood-gen/lib/workshop.ml (hunk 2 above)
Comment (patricoferris) [reply src/ood-gen/lib/workshop.ml]: Yep you are right

File: data/workshops/en/2017.md (hunk 1 above)
Comment (patricoferris) [reply data/workshops/en/2017.md]: Sure thing

**Comment by patricoferris** [comment]
Hopefully fixed everything, @tmattio and I discussed the `ocaml/ocaml.org-media` repository which would add 100M+ to this one I think so we're going to look at CDN solutions for all of the media now that we've hit this bottleneck (the crunched media file would have been too big for Github!). In the short-term I've just added links to that repo for the various slides and pdfs. Note this PR isn't meant to be perfect, just setup the repo ready for someone to go through all the `.md` files and do a better job :))

**Pull request merged**